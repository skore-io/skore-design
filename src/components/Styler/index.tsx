import React from "react";
import styled from "styled-components";

interface Props {
  children: React.ReactChild;
}

const Container = styled.div`
  @import url("https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap");
  align-items: center;
  display: flex;
  font-family: "Poppins", sans-serif;
  justify-content: center;
`;

const Styler: React.FC<Props> = ({ children }) => {
  return <Container>{children}</Container>;
};

export default Styler;
