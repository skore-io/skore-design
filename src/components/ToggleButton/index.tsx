import * as React from "react";
import styled from "styled-components";

export interface ToggleButtonProps {
  /**
   * Click event
   */
  active: boolean;
  /**
   * Color variations
   */
  colorVariation?:
    | "#26E1CD"
    | "#FD4274"
    | "#FECF5C"
    | "#FDAFBB"
    | "#B6B0FC"
    | "#7DDFC3";
  /**
   * Label
   */
  label: string;
  /**
   * Click event
   */
  onClick?: () => void;
  /**
   * Hover event
   */
  onMouseOver?: () => void;
}

const Container = styled.li<ToggleButtonProps>`
  background: ${(props) => props.active && props.colorVariation};
  border: 2px solid ${(props) => props.colorVariation};
  border-radius: 30px;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-weight: 600;
  overflow: hidden;
  padding: 10px 20px;
  user-select: none;
  transition: all 0.15s cubic-bezier(0, 0.5, 1, 0.5);

  &:hover {
    background: ${(props) => props.colorVariation};
  }
`;

/**
 * Bot√£o de Toggle
 */
export const ToggleButton: React.FC<ToggleButtonProps> = (props) => {
  const { label } = props;

  return <Container {...props}>{label}</Container>;
};
