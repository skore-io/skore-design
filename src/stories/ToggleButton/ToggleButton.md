# Title

### Tipo de arquivo ".md"