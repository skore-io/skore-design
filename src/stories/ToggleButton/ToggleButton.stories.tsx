import * as React from "react";
import { Story, Meta } from "@storybook/react/types-6-0";
import { ToggleButton, ToggleButtonProps } from "../../index";
import markdown from "./ToggleButton.md";

export default ({
  title: "SkoreUI/ToggleButton",
  component: ToggleButton,
  argTypes: {
    active: {
      control: "boolean",
    },
    colorVariation: {
      control: {
        type: "inline-radio",
        options: [
          "#26E1CD",
          "#FD4274",
          "#FECF5C",
          "#FDAFBB",
          "#B6B0FC",
          "#7DDFC3",
        ],
      },
    },
    label: "Toggle Button",
    onClick: { action: "clicked" },
    onMouseOver: { action: "hover" },
  },
  parameters: {
    notes: { markdown },
    backgrounds: {
      default: "Dark",
    },
  },
} as unknown) as Meta;

const Template: Story<ToggleButtonProps> = (args) => <ToggleButton {...args} />;

export const Selected = Template.bind({});
Selected.args = {
  active: true,
  colorVariation: "#26E1CD",
  label: "Selected",
};

export const Deselected = Template.bind({});
Deselected.args = {
  active: false,
  colorVariation: "#26E1CD",
  label: "Deselected",
};
