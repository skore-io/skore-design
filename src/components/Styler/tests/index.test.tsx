import * as React from "react";
import { shallow } from "enzyme";
import Styler from "../index";
import { ToggleButton } from "../../ToggleButton";

const component = shallow(
  <Styler>
    <ToggleButton active colorVariation="#FDAFBB" label="BotÃ£o teste" />
  </Styler>
);

describe("ToggleButton", () => {
  it("renders", () => {
    expect(component.exists()).toEqual(true);
  });

  // eslint-disable-next-line jest/no-commented-out-tests
  // it("snapshotShot", () => {
  //   expect(component.getElements()).toMatchSnapshot();
  // });
});
