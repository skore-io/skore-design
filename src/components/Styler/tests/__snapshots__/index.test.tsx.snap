// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton snapshotShot 1`] = `
Array [
  <Styler>
    <styled.li
      active={true}
      colorVariation="#FDAFBB"
      label="Botão teste"
    >
      Botão teste
    </styled.li>
  </Styler>,
]
`;
